src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -1720,12 +1720,52 @@
   <AbstractNode Name="PatternSyntax" Base="CSharpSyntaxNode" />
   <Node Name="DeclarationPatternSyntax" Base="PatternSyntax">
     <Kind Name="DeclarationPattern" />
-    <Field Name="Type" Type="TypeSyntax" />
+    <Field Name="Type" Type="TypeSyntax"/>
     <Field Name="Designation" Type="VariableDesignationSyntax">
       <Kind Name="SingleVariableDesignation"/>
       <Kind Name="DiscardDesignation"/>
     </Field>
   </Node>
+  <Node Name="DeconstructionPatternSyntax" Base="PatternSyntax">
+    <Kind Name="DeconstructionPattern" />
+    <Field Name="Type" Type="TypeSyntax" Optional="true" />
+    <Field Name="OpenParenToken" Type="SyntaxToken">
+      <Kind Name="OpenParenToken"/>
+    </Field>
+    <Field Name="SubPatterns" Type="SeparatedSyntaxList&lt;SubpatternElementSyntax&gt;"/>
+    <Field Name="CloseParenToken" Type="SyntaxToken">
+      <Kind Name="CloseParenToken"/>
+    </Field>
+    <Field Name="PropertySubpattern" Type="PropertySubpatternSyntax" Optional="true" />
+    <Field Name="Designation" Type="VariableDesignationSyntax" Optional="true">
+      <Kind Name="SingleVariableDesignation"/>
+      <Kind Name="DiscardDesignation"/>
+    </Field>
+  </Node>
+  <Node Name="SubpatternElementSyntax" Base="CSharpSyntaxNode">
+    <Kind Name="SubpatternElement"/>
+    <Field Name="NameColon" Type="NameColonSyntax" Optional="true" />
```

Comment:
><Field Name="NameColon" Type="NameColonSyntax" Optional="true" /> [](start = 4, length = 65)

It looks like this doesn't match anything from [https://github.com/dotnet/csharplang/blob/master/proposals/patterns.md](https://github.com/dotnet/csharplang/blob/master/proposals/patterns.md)

Reply:
Correct. This is not tracking the old proposed spec written during the C# 7 timeframe. LDM issues for this iteration are tracked at https://github.com/dotnet/csharplang/issues/1054